lightning/src/ln/peer_handler.rs

```diff
@@ -737,44 +755,54 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 
 	fn do_attempt_write_data(&self, descriptor: &mut Descriptor, peer: &mut Peer) {
 		while !peer.awaiting_write_event {
-			if peer.should_buffer_gossip_backfill() {
-				match peer.sync_status {
-					InitSyncTracker::NoSyncRequested => {},
-					InitSyncTracker::ChannelsSyncing(c) if c < 0xffff_ffff_ffff_ffff => {
-						if let Some((announce, update_a_option, update_b_option)) =
-							self.message_handler.route_handler.get_next_channel_announcement(c)
-						{
-							self.enqueue_message(peer, &announce);
-							if let Some(update_a) = update_a_option {
-								self.enqueue_message(peer, &update_a);
+			if peer.should_buffer_message_backfill() {
+				let next_onion_message_opt = if let Some(peer_node_id) = peer.their_node_id {
+					self.message_handler.onion_message_handler.next_onion_message_for_peer(peer_node_id)
+				} else { None };
+
+				// For now, let onion messages starve gossip.
+				if let Some(next_onion_message) = next_onion_message_opt {
+					self.enqueue_message(peer, &next_onion_message);
+				} else {
+					match peer.sync_status {
+						InitSyncTracker::NoSyncRequested => {},
+						InitSyncTracker::ChannelsSyncing(c) if c < 0xffff_ffff_ffff_ffff => {
```

Comment:
is there some way to abstract this into send_om and then send_init_sync_gossip (if no om was sent)?

Reply:
Let me know what you think of the latest version, sorry for all the churn

Reply:
i think it looks much cleaner now :)